### DISCUSSION TITLE
Optimization Failed

### DISCUSSION THREAD

**Discussion by stefanosbaratsas** [project-pareto/project-pareto#297] [Q&A]
I installed the PARETO framework (v23.12.06 PARETO v0.9.0) on a Windows 11 laptop, and I tried to run the example input file "strategic_permian_demo", but it gives me the following error: 

Optimization Failed
Error: "Index '('K01', 'I0')' is not valid for indexed component 'p_delta_Disposal'"

The data input part looks ok, with the data being populated. 

The same thing happens with the example workshop input file "workshop_baseline_all_data".  

Has anybody else experiences the same issue? Any ideas, how to fix it?

Thanks!

**Comment by tarnold17** [comment]
Hi @stefanosbaratsas - thanks for your question. The cause of the issue you are encountering is that in #281, we updated the parameter `p_delta_Disposal` (which is populated by the input tab "DisposalCapacityIncrements") to be indexed by individual disposal wells (e.g., "K01") in addition to disposal capacity increments (e.g., "I0"):

![image](https://github.com/project-pareto/project-pareto/assets/26798258/dfc4e83e-2f02-4ee4-8ada-eb52479cc670)

The version of the input file you are using presumably looks like the above image. However, the version of the PARETO GUI that you are using is older; it was built before #281 was merged, and so it expects the tab "DisposalCapacityIncrements" to only be indexed by disposal capacity increments. You will need to use an input file where the "DisposalCapacityIncrements" tab looks like this instead:

![image](https://github.com/project-pareto/project-pareto/assets/26798258/7fe8ec24-baa0-42d3-bcda-ea1b2f1beee3)

You can find and download older versions of the input files that should work for you by following this link: https://github.com/project-pareto/project-pareto/tree/b9ddce2d18a2bbb210d743883a7fe9ea9d28e801/pareto/case_studies

I am guessing you might have obtained the input file you used by clicking one of the download buttons in PARETO UI. With the version of the GUI that you are using, the download buttons simply download the latest version of the input file from the main branch of the project-pareto repo. This is how you obtained a version of the input file that is out of date with what your GUI actually expects. We have recently released a new build of PARETO GUI which corrects this behavior - you can download it [here](https://www.project-pareto.org/software/). This new version of the GUI ensures that the download buttons give the user versions of the input files which are compatible with the application. Note that even though the behavior of the download buttons has been fixed in this latest version of the GUI, the backend of this application still uses the older version of the PARETO model which expects the singly-indexed `p_delta_Disposal` and "DisposalCapacityIncrements" tab. In the coming months, we will release a version of the PARETO GUI that uses the newer backend, with the doubly-indexed `p_delta_Disposal` and "DisposalCapacityIncrements" tab.

Our apologies for the error and the confusion. Please don't hesitate to reach out if you have any further questions!
